psi4/src/psi4/libmints/wavefunction.h

```diff
@@ -403,6 +412,20 @@ class PSI_API Wavefunction : public std::enable_shared_from_this<Wavefunction>
     /// Returns the core Hamiltonian matrix
     SharedMatrix H() const { return H_; }
 
+    /// Sets the alpha electrons MO coefficients
+    void set_Ca(SharedMatrix& Ca_new);
+    /// Sets the beta electrons MO coefficients
+    void set_Cb(SharedMatrix& Cb_new);
+    /// Sets the T1 amplitudes
+    virtual void set_T1(SharedMatrix& T1_new) { throw PSIEXCEPTION("There are no T1 amplitudes for this wavefunction."); };
```

Comment:
I dont think should go on the base Wavefunction. Please either 1) add them to the CC* Wavefunctions or 2) (recommend) use the `get/set_array` values on the wavefunction itself which can set/get arbitrary arrays.

Reply:
Ah, right, they can also be set according to 1) this way in `export_wavefunction.cc`. I thought that it is only possible to expose there on the base Wavefunction. :+1: for adding them to the CC*Wavefunctions where needed.

I'm not sure, I understand the `get/set_array` machinery fully: Would 2) mean, that T1/T2 setting in memory would set it in the `array_`  map instead of a direct property of the wavefunctions?
That should work as well, yet it seems to be not heavily used yet in other parts in the code (e.g. how can you access Ca/Cb with this approach), which is why I would prefer option 1) above.